### PR TITLE
feature: S3C-4460 Probe server for Status Proc

### PR DESCRIPTION
**Pull request by miniscruff**
* Refactor create probe server to be reusable
* Queue pop and Queue processor use new create probe server
* Add handle liveness and handle metrics to status proc

### CODE REVIEW

**Comment by miniscruff** [comment]
I am ignoring the conflicts until I get some reviews/check test status to not repeat cycles.

**Review by naren-rajendran — APPROVED** [review]

File: extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js [hunk 1]
```diff
@@ -183,6 +192,78 @@ class ReplicationStatusProcessor {
...
+    }
+
+    /**
+     * Handle ProbeServer metrics
+     *
+     * @param {http.HTTPServerResponse} res - HTTP Response to respond with
+     * @param {Logger} log - Logger
+     * @returns {string} Error response string or undefined
+     */
+    handleMetrics(res, log) {
```
Comment (miniscruff) [inline extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js:260]: no actual metrics reported yet, next PR

File: extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js [hunk 2]
```diff
@@ -183,6 +192,78 @@ class ReplicationStatusProcessor {
...
+                status: constants.statusNotReady,
+            });
+        } else {
+            verboseLiveness.failedCRRProducer = constants.statusReady;
+        }
+
+        log.debug('verbose liveness', verboseLiveness);
+
+        if (responses.length > 0) {
+            return JSON.stringify(responses);
```
Comment (naren-scality) [inline extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js:245]: won't `return` here break?

Comment (miniscruff) [reply extensions/replication/replicationStatusProcessor/ReplicationStatusProcessor.js:245]: Currently the arsenal probe server will handle string responses and finish the response for you ( it has been proposed to change that but that is what it is currently doing )

**Review by saitharunthodupunuru — APPROVED** [review]

**Comment by miniscruff** [comment]
Resolving conflicts and approving

**Pull request merged**